File: java/Mediator/src/Buyer.java
package Exercise;

public class Buyer {
	Mediator mediator;
	String currency;

	public Buyer(Mediator mediator, String currency) {
		super();
		this.mediator = mediator;
		this.currency = currency;
		mediator.____________(__________); // <<1 <<2 
	}

	public boolean attemptToPurchase(float bid) {
		System.out.println("Buyer attempting a bid of " + currency +" " bid + " .");
		return mediator.__________(__________, currency); // <<3 <<4
	}
	
}

File: ruby/Mediator/buyer.rb


class Buyer
  attr_accessor :mediator
  attr_accessor :currency

  def initialize(mediator,currency)
    @mediator = mediator
    @currency = currency
    mediator.____________(____________) # <<1 <<2
  end

  def attemptToPurchase(bid)
    print "Buyer attempting a bid of #{currency} #{bid}.\n"
    return mediator.____________(____________, currency) # <<3 <<4
  end
end